src/distilabel/pipeline.py

```diff
@@ -439,14 +504,20 @@ def _build_dataset(  # noqa: C901
         _dataset.task = self.labeller.task if self.labeller is not None else None  # type: ignore
         return _dataset  # type: ignore
 
+    def _teardown(self) -> None:
+        if self.generator is not None and isinstance(self.generator, ProcessLLM):
+            self.generator.teardown()
+
+        if self.labeller is not None and isinstance(self.labeller, ProcessLLM):
+            self.labeller.teardown()
+
     def _generate(  # noqa: C901
         self,
         dataset: Dataset,
         num_generations: int = 1,
         batch_size: int = 1,
         enable_checkpoints: bool = True,
         display_progress_bar: bool = False,
-        verbose: bool = True,
     ) -> CustomDataset:
```

Comment:
Why did we remove the `verbose` arg?

Reply:
Logging can be configured using `DISTILABEL_LOG_LEVEL` environment variable and I think that the `logger` should be configured just in one place in the code, not in several, as it can be confusing.